driver/modern_bpf/CMakeLists.txt

```diff
@@ -6,22 +6,89 @@ else()
   set(DEBUG "")
 endif()
 
+########################
+# Check kernel version.
+########################
+
+# We check it here because the skeleton could be provided with the `MODERN_BPF_SKEL_DIR` env variable
+# so the compilation is still possible on older kernels.
+execute_process(COMMAND uname -r OUTPUT_VARIABLE UNAME_RESULT OUTPUT_STRIP_TRAILING_WHITESPACE)
+message(-- " Kernel version: " ${UNAME_RESULT})
+string(REGEX MATCH "[0-9]+.[0-9]+" LINUX_KERNEL_VERSION ${UNAME_RESULT})
+
+if (LINUX_KERNEL_VERSION VERSION_LESS 5.8)
+    message(FATAL_ERROR "Linux kernel version should be >= 5.8 but actual kernel version is: ${UNAME_RESULT}")
+endif()
+
+# This is needed to obtain the modern bpf driver version
 include(compute_versions RESULT_VARIABLE RESULT)
 if(RESULT STREQUAL NOTFOUND)
   message(FATAL_ERROR "problem with compute_versions.cmake in ${CMAKE_MODULE_PATH}")
 endif()
 compute_versions(../API_VERSION ../SCHEMA_VERSION)
 configure_file(../driver_config.h.in ${CMAKE_CURRENT_SOURCE_DIR}/../driver_config.h)
 
-#### TODO: we could add checks on the minimum required clang version.
-find_program(CLANG_EXE NAMES clang DOC "Path to clang executable")
+########################
+# Check clang version.
+########################
+
+# If the user doesn't provide the path to `clang` try to find it locally
+if(NOT MODERN_CLANG_EXE)
+  find_program(MODERN_CLANG_EXE NAMES clang DOC "Path to clang executable")
+  if(NOT MODERN_CLANG_EXE)
+    message(FATAL_ERROR "unable to find clang")
+  endif()
+endif()
+
+# Check the right clang version (>=12)
+execute_process(COMMAND ${MODERN_CLANG_EXE} --version
+  OUTPUT_VARIABLE CLANG_version_output
+  ERROR_VARIABLE CLANG_version_error
+  RESULT_VARIABLE CLANG_version_result
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+if(${CLANG_version_result} EQUAL 0)
+  if("${CLANG_version_output}" MATCHES "clang version ([^\n]+)\n")
+    # Transform X.Y.Z into X;Y;Z which can then be interpreted as a list
+    set(CLANG_VERSION "${CMAKE_MATCH_1}")
+    string(REPLACE "." ";" CLANG_VERSION_LIST ${CLANG_VERSION})
+    list(GET CLANG_VERSION_LIST 0 CLANG_VERSION_MAJOR)
+
+    # Anything older than clang 12 doesn't really work
+    string(COMPARE LESS ${CLANG_VERSION_MAJOR} 12 CLANG_VERSION_MAJOR_LT12)
+
+    if(${CLANG_VERSION_MAJOR_LT12})
+      message(FATAL_ERROR "clang ${CLANG_VERSION} is too old for compiling the modern BPF probe")
+    endif()
+
+    message(STATUS "Found clang version: ${CLANG_VERSION}")
+  else()
+    message(FATAL_ERROR "Failed to parse clang version string: ${CLANG_version_output}")
+  endif()
+else()
+  message(FATAL_ERROR "Command \"${MODERN_CLANG_EXE} --version\" failed with output:\n ${CLANG_version_error}")
+endif()
+
+message(STATUS "clang used by the modern bpf probe: '${MODERN_CLANG_EXE}'")
+
+########################
+# Check bpftool version.
+########################
```

Comment:
@Andreagit97 really nice to see this PR up, thanks a lot!

Would we also want to check the actual `bpftool` version? :upside_down_face: 
Both formats, depending on whether you built it from the kernel tree or libbpf.

Reply:
great idea, done :)